src/pyodide/internal/emscriptenSetup.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * This file is intended to be executed in the Python pool (once it exists). As such, it cannot
+ * import anything that transitively uses C++ extension modules. It has to work in a vanilla v8
+ * isolate. Also, we will have to bundle this file and all of its transitive imports into a single
+ * js file.
+ */
+
+import { reportError } from 'pyodide-internal:util';
+
+/**
+ * _createPyodideModule and pyodideWasmModule together are produced by the
+ * Emscripten linker
+ */
+import { _createPyodideModule } from 'pyodide-internal:generated/pyodide.asm';
+
+/**
+ * A preRun hook. Make sure environment variables are visible at runtime.
+ */
+function setEnv(Module: Module): void {
+  Object.assign(Module.ENV, Module.API.config.env);
+}
+
+function getWaitForDynlibs(resolveReadyPromise: PreRunHook): PreRunHook {
```

Comment:
Nit: this doesn't appear to wait for anything, rather it returns a function that resolves a promise. Should the name be changed to better reflect this?

Reply:
By calling `addRunDependency("dynlibs")` it causes Emscripten to wait until we call `removeRunDependency("dynlibs")`. I will add a comment explaining this since it is indeed a bit opaque.

Reply:
Oh there already is a comment for it.